wallet-sdk/src/main/kotlin/org/stellar/walletsdk/util/Format.kt

```diff
@@ -121,6 +127,223 @@ suspend fun formatAccountBalances(account: AccountResponse, server: Server): For
+            }
+          } else {
+            operation.from
+          }
+        )
+        .asset(formatWalletAsset(operation))
+        .type(
+          if (isSender) {
+            if (isSwap) {
+              WalletOperationType.SWAP
+            } else {
+              WalletOperationType.SEND
+            }
+          } else {
+            WalletOperationType.RECEIVE
+          }
+        )
+        .build()
+    }
+    // Other
+    else -> {
+      return opBuilder.build()
+    }
+  }
+}
+
+/**
+ * Format anchor transactions to make them consistent.
+ *
+ * @param transaction anchor transaction to format
+ * @param asset transaction asset
+ *
+ * @return formatted transaction
+ */
+fun formatAnchorTransaction(
+  transaction: AnchorTransaction,
+  asset: Asset
+): WalletOperation<AnchorTransaction> {
+  val opBuilder = WalletOperationBuilder<AnchorTransaction>().fromTransaction(transaction, asset)
+
+  when (transaction.kind) {
+    AnchorTransactionType.DEPOSIT.type,
+    AnchorTransactionType.WITHDRAW.type -> {
+      return opBuilder
+        .amount(transaction.amount_out)
+        .type(
+          if (transaction.kind == AnchorTransactionType.DEPOSIT.type) {
+            WalletOperationType.DEPOSIT
+          } else {
+            WalletOperationType.WITHDRAW
+          }
+        )
+        .build()
+    }
+    else -> {
+      return opBuilder.build()
+    }
+  }
+}
+
+/**
+ * Helper class to format wallet operation from Stellar operation or anchor transaction.
+ *
+ * @param T type of operation or transaction
+ */
+internal class WalletOperationBuilder<T : Any>() {
+  lateinit var id: String
+  lateinit var date: String
+  lateinit var amount: String
+  lateinit var account: String
+  lateinit var asset: List<WalletAsset>
+  lateinit var type: WalletOperationType
+  lateinit var rawOperation: T
+
+  internal fun defaults() = apply {
+    this.amount = ""
+    this.account = ""
+    this.type = WalletOperationType.OTHER
+  }
+
+  fun id(id: String) = apply { this.id = id }
+  fun date(date: String) = apply { this.date = date }
+  fun amount(amount: String) = apply { this.amount = amount }
+  fun account(account: String) = apply { this.account = account }
+  fun asset(asset: List<WalletAsset>) = apply { this.asset = asset }
+  fun type(type: WalletOperationType) = apply { this.type = type }
+  fun rawOperation(rawOperation: T) = apply { this.rawOperation = rawOperation }
+
+  fun build(): WalletOperation<T> {
+    return WalletOperation(
+      id = this.id,
+      date = this.date,
+      amount = this.amount,
+      account = this.account,
+      asset = this.asset,
+      type = this.type,
+      rawOperation = this.rawOperation as T
+    )
+  }
+}
+
+internal fun <T : OperationResponse> WalletOperationBuilder<T>.fromOperation(operation: T) = apply {
+  this.defaults()
+  this.id = operation.id.toString()
+  this.date = operation.createdAt
+  this.asset = listOf()
+  this.rawOperation = operation
+}
```

Comment:
```suggestion

  companion object {
    internal fun <T : OperationResponse> fromOperation(operation: T): WalletOperationBuilder<T> {
      return WalletOperationBuilder<T>().apply {
        this.defaults()
        this.id = operation.id.toString()
        this.date = operation.createdAt
        this.asset = listOf()
        this.rawOperation = operation
      }
    }
  }
```
More common approach (factory function). 
Then it can be used without creating builder
```kotlin
val opBuilder = WalletOperationBuilder.fromOperation(operation)
````